### DISCUSSION TITLE
Request to add anti aliasing feature

### DISCUSSION THREAD

**Discussion by 614386079** [ikemen-engine/Ikemen-GO#1420] [Feature Request] [answered]
### Is your feature request related to a problem? Please describe.

Low definition images have severe aliasing. If there is anti aliasing function, ikemen go will be better

**Comment by SuperFromND** [comment, marked as answer]
I beleive we already implement MSAA; it can be enabled in the options menu.

**Reply by 614386079** [reply]
I turned on MSAA, but it didn't produce any anti aliasing effect

**Reply by SuperFromND** [reply]
![image](https://github.com/ikemen-engine/Ikemen-GO/assets/22881403/ba4837f7-4a24-4516-821b-e60c10b012dc)
Tested on 0.98.2 and it certainly does seem to work on scaled assets like stages with zoom. (see Eggman's sprite in the middle there):
![image](https://github.com/ikemen-engine/Ikemen-GO/assets/22881403/8b548cbb-e438-4137-a2a9-b02ff007959a)

Maybe you actually mean *bilinear* filtering, like this?
![image](https://github.com/ikemen-engine/Ikemen-GO/assets/22881403/9fb5eb4a-e735-4567-bd79-3b358c322f3c)

**Reply by 614386079** [reply]
Okay, I understand

**Comment by shinra358** [comment]
we need xbrz up to 6. reshade works differently on this engine probably because of the different elements having different rezzes. we cant achieve vector imaging like in real games or emus without making text and lifebars smudgy.

it would be great to have a sctrl that allows stage elements in the stages and certain character elements in characters be filtered with whatever filter is chosing because not all elements in the scene would need filtering bcuz somethings will be highres already. but this will help w any jaggy element without affect text and lifebars:

type=filter
trigger1=
mode= (Xbrz, etc)
value=(strength of chosen mode)

msaa wont do just for low rez chars, msaa only prevents shimmering of edges. it doesnt turn deep pixelation into a smooth line.
